internal/plans/plan_test.go

```diff
@@ -0,0 +1,56 @@
+package plans
+
+import (
+	"bytes"
+	"testing"
+
+	root "github.com/equinix/metal-cli/internal/cli"
+	outputPkg "github.com/equinix/metal-cli/internal/outputs"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCli_Plans(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping test in short mode.")
+	}
+	subCommand := "plans"
+	consumerToken := ""
+	apiURL := ""
+	Version := "metal"
+	rootClient := root.NewClient(consumerToken, apiURL, Version)
+	type fields struct {
+		MainCmd  *cobra.Command
+		Outputer outputPkg.Outputer
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    *cobra.Command
+		cmdFunc func(*testing.T, *cobra.Command)
+	}{
+		{
+			name: "get",
+			fields: fields{
+				MainCmd:  NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				out := &bytes.Buffer{}
+				root.SetArgs([]string{subCommand, "get"})
+				root.SetOutput(out)
+				err := root.Execute()
+				assert.NoError(t, err)
```

Comment:
I think the `plans` subcommand warrants more detailed testing.  For example:  it would be useful to also have a test that uses relevant command line flags and confirms that the response has the expected data (e.g., if I specify a filter `slug=m3.small.x86`, I get back plan data that has a `slug` of `m3.small.x86`).

Reply:
updated